code/lib/components/src/icon/icon.tsx

```diff
@@ -1,39 +1,54 @@
-import React, { ComponentProps, memo } from 'react';
+import React, { FunctionComponent, ComponentProps, memo } from 'react';
 import { styled } from '@storybook/theming';
-import icons, { IconKey } from './icons';
+import { icons, IconKey } from './icons';
 
-import Svg from './svg';
+const Svg = styled.svg`
+  display: inline-block;
+  shape-rendering: inherit;
+  transform: translate3d(0, 0, 0);
+  vertical-align: middle;
 
-const Path = styled.path({
-  fill: 'currentColor',
-});
+  path {
+    fill: currentColor;
+  }
+`;
 
-export interface IconsProps extends ComponentProps<typeof Svg> {
-  icon?: IconKey;
-  symbol?: IconKey;
+/**
+ * An Icon is a piece of visual element, but we must ensure its accessibility while using it.
+ * It can have 2 purposes:
+ *
+ * - *decorative only*: for example, it illustrates a label next to it. We must ensure that it is ignored by screen readers, by setting `aria-hidden` attribute (ex: `<Icon icon="check" aria-hidden />`)
+ * - *non-decorative*: it means that it delivers information. For example, an icon as only child in a button. The meaning can be obvious visually, but it must have a proper text alternative via `aria-label` for screen readers. (ex: `<Icon icon="print" aria-label="Print this document" />`)
+ */
+export interface IconsProps {
```

Comment:
Was the `ComponentProps<typeof Svg>` left off intentionally? Seems useful to accept those props given that they are spread on the root `<svg>`.

Reply:
It wasn't. I copied the code from the Icon Component in the Design System and merged it in here. I would be happy to implement that with a little guidance.